File: cvi-web/src/app/topics/dto/topic-create-dto.ts
export interface TopicCreateDto {
  path: string;
  protocol: string;
  middlewareEndpoint: string;
  topicType: string;
  dataType: string;
  hardwareType: string;
  unit: string;
  messageFormat: string;
  metaModelType: string;
  metaModel: string;
  country: string;
  state: string;
  city: string;
  street: string;
  point: string;
}

File: entities-management/src/main/java/de/uni/stuttgart/ipvs/tm/topics/dto/TopicCreateDTO.java
package de.uni.stuttgart.ipvs.tm.topics.dto;

import lombok.Data;

import de.uni.stuttgart.ipvs.tm.topics.properties.HardwareProperties;
import de.uni.stuttgart.ipvs.tm.topics.properties.LocationProperties;
import de.uni.stuttgart.ipvs.tm.topics.properties.MessageProperties;
import de.uni.stuttgart.ipvs.tm.topics.properties.TopicProperties;

@Data
public class TopicCreateDTO
        implements TopicProperties, MessageProperties, HardwareProperties, LocationProperties {

    private String path;
    private String middlewareEndpoint;
    private String dataType;
    private String protocol;
    private String topicType;

    private String hardwareType;
    private String unit;

    private String messageFormat;
    private String metaModelType;
    private String metaModel;

    private String country;
    private String state;
    private String city;
    private String street;
    private String point;
}

